Fix build and add alias targets

lrewega opened the pull request.

lrewega committed:
Update libsodium URL

libsodium relocated some releases to an "old" directory. Let's correct
the URL to point to the new location of the 1.0.11 release.

lrewega committed:
Add aliased "install" and "uninstall" targets

Existing Vertica UDx projects use "install" and "uninstall" targets to
register plugins and execute DDL. Let's add aliases to conform with
those projects.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/vertica-aesgcm-udx?pullRequest=1) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/vertica-aesgcm-udx?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/vertica-aesgcm-udx?pullRequest=1) it.</sub>

lrewega committed:
Add support for libsodium download mirrors

To future proof against URL changes, let's try to download libsodium
from multiple providers.

lrewega committed:
Fix distclean to remove libsodium directory

The distclean target mistakenly attempted to remove the libsodium
directory non-recursively.

lrewega committed:
Upgrade libsodium to 1.0.15
